### ISSUE TITLE
[java] UnusedModifier doesn't check annotations

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#246]
**Rule Set:** [unused/UnusedModifier](https://pmd.github.io/pmd-5.5.3/pmd-java/rules/java/unusedcode.html#UnusedModifier)

**Description:**
Annotation elements are `public abstract` by default, no other modifiers are allowed. Explicitly setting them matches this rule's criteria:

> For historical reasons, modifiers which are implied by the context are accepted by the compiler, but are superfluous.

The same applies to fields on annotations, which are `public static final` by default.

**Code Sample demonstrating the issue:**

```java
public @interface Test {
  public static final String DEFAULT_MESSAGE = "message";
  public abstract String message() default DEFAULT_MESSAGE;
}
```

PMD will find no violations on this annotation, 5 violations are expected.

**Running PMD through:** Any

P.S. This class should probably go to the `unnecessary` ruleset rather the `unused` one... even the unit tests' description usually start with *"Unnecessary ..."*
